### DISCUSSION TITLE
Support config setSseTabs

### DISCUSSION THREAD

**Discussion by MeiyongYu** [dail8859/NotepadNext#644] [Ideas]
At tabs or space the diff company require difference.
Some require tabs, the famous is linux, and use tab is trend.
Also many one like space, because in any environment it will not confusion.

I have add a setUseTabs config in setting.
When the flag is set, all editor support the config.

setUseTabs result:
![image](https://github.com/user-attachments/assets/9480d0b1-aa1a-442b-a4b0-792ef372d998)
![image](https://github.com/user-attachments/assets/5f7dbeed-ff46-4b24-b1e0-f3093834d420)

not setUseTabs result:
![image](https://github.com/user-attachments/assets/4e4a1e7b-922e-4cb7-a367-6dc7fbbdb767)
![image](https://github.com/user-attachments/assets/4b1e23b0-1810-4fed-ade3-5b055bd71245)

the PR is ===> https://github.com/dail8859/NotepadNext/pull/642

**Comment by dail8859** [comment]
I think if this is implemented it needs to be clear somehow this is a **global** override.

For example, Notepad++ does it this way to set a default:

![image](https://github.com/user-attachments/assets/193f65f6-aef6-4599-b127-95427ca559cc)


And then allows individual languages to override the default:

![image](https://github.com/user-attachments/assets/52691acc-faad-435e-b135-0492151c9b64)


I'm not suggesting that Notepad++ should be used as the exact way to implement it, but only as an example of how it shows different languages can override it.

**Reply by MeiyongYu** [reply]
OK. I will add new config like notepad++.

**Reply by dail8859** [reply]
Part of the problem is currently there's no way for the language settings (in the lua files) to be modified at all yet. So simply if the change you have so far is just renamed to a "global override" to allow the user to either pick tabs or spaces would probably be a good solution.

**Reply by MeiyongYu** [reply]
1) The setting page like this:
![image](https://github.com/user-attachments/assets/38c61733-7748-4321-8402-4b4734566570)

2) And in edit menu, add a control clicking to temporary to change the current file to tabs or space mode, like this:
   
![image](https://github.com/user-attachments/assets/6fa00338-7951-4b8d-b56b-0eda843a95a7)

And next time i can base this menu, add "convert select text to tab" and  "convert select text to space" command

**Reply by dail8859** [reply]
> The setting page like this:

That looks better, but probably need to provide a way to for the user to select either tabs, spaces, or not select anything at all. This is really only a temporary solution anyways until individual languages can bet modified permanently by the user.

> And in edit menu, add a control clicking to temporary to change the current file to tabs or space mode, like this

There should probably be some way for the user to select what they want to user, and potentially converting between tabs and spaces, but that is obviously more involved.

**Reply by MeiyongYu** [reply]
> > The setting page like this:
> 
> That looks better, but probably need to provide a way to for the user to select either tabs, spaces, or not select anything at all. 

If user neigthor select tabs and space, what the behavior is the software.

This is really only a temporary solution anyways until individual languages can bet modified permanently by the user.
> 
> > And in edit menu, add a control clicking to temporary to change the current file to tabs or space mode, like this
> 
> There should probably be some way for the user to select what they want to user, and potentially converting between tabs and spaces, but that is obviously more involved.

Yes, if global setting is set, but when edit some file have to set anthor mode temporary.

**Comment by dail8859** [comment]
> If user neigthor select tabs and space, what the behavior is the software.

Each language is configured via a lua script and they can specify if the default should be tabs or spaces. E.g:

https://github.com/dail8859/NotepadNext/blob/4c7934479b01378d81acd4652e79846c685ebdbc/src/NotepadNext/languages/python.lua#L9

So if nothing is selected it should use whatever is set in the Lua script which it currently does. If the lua script doesn't specify anything then it should not be set.
